src/bootstrap/healthcheck.js

```diff
@@ -0,0 +1,96 @@
+import network from "../store/selectors/network";
+import {
+  setLastNetworkCheckEndAt,
+  setLastNetworkCheckStartAt
+} from "../store/reducers/network/onNetworkCheckEndEpic";
+import platformReducer from "../store/reducers/platform";
+import networkReducer from "../store/reducers/network";
+import * as Network from "./network";
+import { checkIfOutOfSyncEpic } from "../store/reducers/network/checkIfOutOfSync";
+import Raven from "raven-js";
+import { registerSubscription } from "../utils/subscriptions/registerSubscription";
+import {
+  HAS_ACCOUNTS,
+  SUBSCRIPTIONS_GROUP_GLOBAL_INITIAL,
+  SUBSCRIPTIONS_LATEST_BLOCK
+} from "../constants";
+import { subscribeLatestBlockFilterEpic } from "../store/reducers/network/subscribeLatestBlockFilterEpic";
+import accounts from "../store/selectors/accounts";
+import accountsReducer from "../store/reducers/accounts";
+import { Session } from "../utils/session";
+import { CheckNetworkAction } from "../store/reducers/network/CheckNetworkAction";
+
+export const healthCheck = async (
+  dispatch,
+  getState,
+  isInitialHealthcheck = false
+) => {
+  if (network.isNetworkCheckPending(getState()) === true) {
+    return;
+  }
+  dispatch(setLastNetworkCheckStartAt());
+  if (isInitialHealthcheck) {
+    dispatch(platformReducer.actions.setGlobalFormLockEnabled());
+    dispatch(networkReducer.actions.connecting());
+  }
+  try {
+    const providerType = await Network.checkConnectivity().catch();
+    const connectedNetworkId = await dispatch(
+      networkReducer.actions.getConnectedNetworkId()
+    );
+    if (isInitialHealthcheck) {
+      await dispatch(checkIfOutOfSyncEpic());
+      // console.log("connectedTo:", network.activeNetworkMeta(getState()).get('name'));
+      Raven.setTagsContext({
+        network: network.activeNetworkMeta(getState()).get("name")
+      });
+    }
+    if (providerType && connectedNetworkId.value) {
+      dispatch(networkReducer.actions.connected());
+      if (isInitialHealthcheck) {
+        /**
+         * We only do this once, since later we subscribe to 'latest' filter
+         * to get notified on new block resolved
+         */
+        dispatch(networkReducer.actions.getLatestBlockNumber());
+        registerSubscription(
+          SUBSCRIPTIONS_LATEST_BLOCK,
+          () => dispatch(subscribeLatestBlockFilterEpic()),
+          { dispatch, getState },
+          SUBSCRIPTIONS_GROUP_GLOBAL_INITIAL
+        );
+      }
+      const previousDefaultAccount = accounts.defaultAccount(getState());
+      if (
+        HAS_ACCOUNTS === await dispatch(accountsReducer.actions.checkAccountsEpic())
+      ) {
+        try {
+          /**
+           *  Initialize session on first run of the healthcheck or when default address changes
+           */
+          if (
+            isInitialHealthcheck || previousDefaultAccount !== accounts.defaultAccount(getState())
```

Comment:
Getting little bit confused in here. So on line 63 you are getting value , then you execute some check ( `checkAccountsEpic`) and then you compare the `previousDefaultAccount` value with returned value from the same function call ` accounts.defaultAccount(..)`.

The only place where I see that something might change the defaultAccount is the `checkAccountsEpic` but if that's the case then the name of the method is wrong. When you check something you don't change it right ?
Also that `checkAccounts` doesn't bring much meaning into what it is supposed to be checking for ?

`accounts.actions.checkIfHasAny` - What do you think about that?